File: src/agent-sec-core/openclaw-plugin/src/demo-coverage.ts
/**
 * Demo module for testing incremental coverage gate.
 *
 * Contains pure utility functions to verify diff-cover behavior.
 */

/** Add two numbers. */
export function add(a: number, b: number): number {
  return a + b;
}

/** Subtract b from a. */
export function subtract(a: number, b: number): number {
  return a - b;
}

/** Multiply two numbers. */
export function multiply(a: number, b: number): number {
  return a * b;
}

/** Safe division — returns null for zero divisor. */
export function divide(a: number, b: number): number | null {
  if (b === 0) {
    return null;
  }
  return a / b;
}

/** Check if number is positive. */
export function isPositive(n: number): boolean {
  if (n > 0) {
    return true;
  }
  return false;
}

/** Check if number is even. */
export function isEven(n: number): boolean {
  if (n % 2 === 0) {
    return true;
  }
  return false;
}

/** Clamp value between low and high. */
export function clamp(value: number, low: number, high: number): number {
  if (value < low) {
    return low;
  }
  if (value > high) {
    return high;
  }
  return value;
}

/** Return letter grade for a score. */
export function grade(score: number): string {
  if (score >= 90) {
    return "A";
  } else if (score >= 80) {
    return "B";
  } else if (score >= 70) {
    return "C";
  } else if (score >= 60) {
    return "D";
  } else {
    return "F";
  }
}

/** Classic fizzbuzz. */
export function fizzbuzz(n: number): string {
  if (n % 15 === 0) {
    return "FizzBuzz";
  } else if (n % 3 === 0) {
    return "Fizz";
  } else if (n % 5 === 0) {
    return "Buzz";
  } else {
    return String(n);
  }
}

/** Calculate factorial (throws for negative). */
export function factorial(n: number): number {
  if (n < 0) {
    throw new Error("Negative numbers not allowed");
  }
  if (n <= 1) {
    return 1;
  }
  return n * factorial(n - 1);
}

File: src/agent-sec-core/linux-sandbox/src/demo_coverage.rs
//! Demo module for testing incremental coverage gate.
//!
//! Contains pure functions (no Linux-specific deps) to verify diff-cover behavior.

/// Add two numbers.
pub fn add(a: i64, b: i64) -> i64 {
    a + b
}

/// Subtract b from a.
pub fn subtract(a: i64, b: i64) -> i64 {
    a - b
}

/// Multiply two numbers.
pub fn multiply(a: i64, b: i64) -> i64 {
    a * b
}

/// Safe division returning None for zero divisor.
pub fn divide(a: i64, b: i64) -> Option<f64> {
    if b == 0 {
        None
    } else {
        Some(a as f64 / b as f64)
    }
}

/// Check if number is positive.
pub fn is_positive(n: i64) -> bool {
    n > 0
}

/// Clamp value between low and high (inclusive).
pub fn clamp(value: i64, low: i64, high: i64) -> i64 {
    if value < low {
        low
    } else if value > high {
        high
    } else {
        value
    }
}

/// Return letter grade for a score.
pub fn grade(score: u32) -> &'static str {
    if score >= 90 {
        "A"
    } else if score >= 80 {
        "B"
    } else if score >= 70 {
        "C"
    } else if score >= 60 {
        "D"
    } else {
        "F"
    }
}

/// Classic fizzbuzz.
pub fn fizzbuzz(n: u32) -> String {
    if n % 15 == 0 {
        "FizzBuzz".to_string()
    } else if n % 3 == 0 {
        "Fizz".to_string()
    } else if n % 5 == 0 {
        "Buzz".to_string()
    } else {
        n.to_string()
    }
}

/// Calculate factorial. Returns None for negative conceptual input (handled via u64).
pub fn factorial(n: u64) -> u64 {
    if n <= 1 {
        1
    } else {
        n * factorial(n - 1)
    }
}
